cni/pkg/nodeagent/server.go

```diff
@@ -157,16 +157,20 @@ func (s *Server) Start() {
 	log.Info("CNI ambient server starting")
 	s.kubeClient.RunAndWait(s.ctx.Done())
 	log.Info("CNI ambient server kubeclient started")
+	// Start the informer handlers FIRST, before we snapshot.
+	// They will keep the (mutex'd) snapshot cache synced.
+	s.handlers.Start()
```

Comment:
This doesn't ensure we finish the initial reconcile (would need s.queue.HasSynced in InformerHandlers.Start). Is that somethign we want

Reply:
We're already doing 

```
func (s *InformerHandlers) Start() {
	kube.WaitForCacheSync("informer", s.ctx.Done(), s.pods.HasSynced, s.namespaces.HasSynced)
	go s.queue.Run(s.ctx.Done())
}
```

Is that not equivalent? If not, I'll add.

Reply:
Not quite.. its all very subtle

* `s.pods.HasSynced` with raw informer (which we don't use): the underlying cache has the info, handlers may or may not be called
* `s.pods.HasSynced` with kclient (which we use): cache synced + handlers called

But the handler action is just "add to the queue"

* `s.queue.HasSynced` - every event that was enqueue before Run was called has completed proecssing

Note we call queue.Run after pods.HasSynced, so that means that `s.queue.HasSynced` means we got all the initial state, added them to the queue, and processed them from the queue.

https://github.com/istio/istio/blob/bd9e52d05817b03b9811a761a818b2e76d356018/pkg/kube/controllers/example_test.go#L45 is a good example covering the subtleties here

Reply:
...actually, now that I think about it - we do not want that. Since we might not have a ztunnel connection yet, and we can't process all events in the queue successfully until we do.